### ISSUE TITLE
Many padic operations set the valuation to zero.

### ISSUE THREAD

**Issue by wbhart** [wbhart/flint2#679]
Hartmut Monien has noticed that when an operation returns zero, the valuation is set to zero, even if the precision is higher than that. This is due to the zero function being called systematically when the computed valuation of the output is bigger than or equal to the precision.

This seems like a deliberate choice, and many tests fail if the behaviour of the zero function is changed. At the very least, this odd choice should be clearly documented, if the behaviour shouldn't be changed altogether.

Here is a simple C program which exhibits it:

```
#include <flint/flint.h>
#include <flint/fmpz.h>
#include <flint/padic.h>
#include <flint/padic_poly.h>

int main() {
  fmpz_t p;
  fmpz_init(p);
  fmpz_set_ui(p, 7);
  
  padic_ctx_t ctx;
  padic_ctx_init(ctx, p, 0, 128, PADIC_SERIES);

  padic_t x, two;
  
  padic_init(x);
  padic_init(two);
  padic_set_ui(two, 2, ctx);
  padic_sqrt(x, two, ctx);
  padic_mul(x, x, x, ctx);
  padic_sub(x, two, x, ctx);
  flint_printf("%ld\n", padic_val(x));
}
```

**Comment by wbhart** [comment]
We've decided to consult with the community and see if anyone relies on the existing behaviour. If not, we've decided to change it so the valuation equals the precision in zero.

**Comment by wbhart** [comment]
I spent some time trying to change over to val = prec when zero. But then I realised that the padic module uses an MPFR-like interface where computations are done exactly and then set to the precision of the output.

In particular, setting the valuation to zero doesn't affect anything because all the functions check for zero by just examining the unit part of the padic, and then the computation is done to the full precision, then set to the precision of the output, just like in every other case. The valuation is not used for such computations and so there is no loss of precision. In fact, the precision of the output is never actually changed!

I think we should leave things as they are for now, as it is quite a lot of work to entirely change the interface, and I don't think the current interface is really broken.
